BaseClasses.py

```diff
@@ -233,6 +233,8 @@ def set_options(self, args: Namespace) -> None:
             options_dataclass: type[Options.PerGameCommonOptions] = world_type.options_dataclass
             self.worlds[player].options = options_dataclass(**{option_key: getattr(args, option_key)[player]
                                                                for option_key in options_dataclass.type_hints})
+            # TODO for back compatibility
```

Comment:
Without a plan for how to deprecate the old way, I think this is not a good a idea.
We'll be maintaining the 2 different ways forever.

Reply:
Yeah this method doesn't even work currently. There was some discussion about it on discord https://discordapp.com/channels/731205301247803413/731214280439103580/1358551158440067212 but I'm still unsure what direction we should take with this, but I still think it's a good idea to do in some way.